packages/block-library/src/file/index.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Server-side rendering of the `core/file` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/file` block on server.
+ *
+ * @param array $attributes The block attributes.
+ * @param array $content    The block content.
+ *
+ * @return string Returns the modified block content.
+ */
+function render_block_core_file( $attributes, $content ) {
+	$script = '';
+	if ( ! empty( $attributes['showInlineEmbed'] ) ) {
+		$script = <<<HTML
```

Comment:
Why is this script not an external script that is set as `WP_Block_Type::$script` when registering the block  using `register_block_type`? Or alternatively, why not make it an external script and call `wp_enqueue_script` if `! empty( $attributes['showInlineEmbed'] )`?

Reply:
Also, as an external script it can be de-duped with `packages/block-library/src/file/utils.js`

Reply:
`wp_enqueue_script()` is probably the best option for de-duplicating the code, but I'm not aware of that method being used for any other blocks. (Or even the `script` property being used, for that matter.)

It seems like it would need a special case to keep it unbundled, and possibly some additional magic when it's merged to Core?

Reply:
2b239f3102 is a brief exploration of this: I don't mind it for this PR, I'm not wild about it as a generic solution.

Reply:
I see `render_callback` as a function that the sole purpose is to render (return) the content of the block and not trigger a side effect elsewhere. Using `wp_enqueue_script` creates an indirection that is very hard to follow.

I think a lot of people ask ways for providing frontend scripts for blocks and it could be a property in `register_block_type` (the equivalent to `editor_script` => `frontend_script`).

For the purpose of this PR specifically, I don't mind the hack, whether it's inline or `wp_enqueue_script`, both are not ideal solution for me but ok.